### PR TITLE
Thycotic

### PR DESCRIPTION
**Pull request by AndyNikolaev**
Fixes #

## Proposed Changes
Reopen PR for Thycotic Secret Server integration
  -
  -
  -

### CODE REVIEW

File: Workbooks/ThycoticWorkbook.json
Comment (v-jayakal) [inline Workbooks/ThycoticWorkbook.json:122]: Please add PROVIDER NAME and APPLIANCE NAME in the queries makes them return appropriate results for the connector.

File: Workbooks/ThycoticWorkbook.json
Comment (v-jayakal) [inline Workbooks/ThycoticWorkbook.json:1]: Change workbooksMetadata.json file, so that it will contain a new section for Thycotic connector workbook. Please refer step 2 of https://github.com/Azure/Azure-Sentinel/blob/30b5ade4aa74942aca543bbb33db2d360e08b810/Workbooks/README.md for more details. You can check the metadata file https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/WorkbooksMetadata.json with metadata of other connectors sections as reference.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: DataConnectors/ThycoticCEF.json
Comment (v-jayakal) [inline DataConnectors/ThycoticCEF.json:9]: Legend should be **_CommonSecurityLog_** , the data type name, Please refer CEF templatehttps://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: DataConnectors/ThycoticCEF.json
Comment (AndyNikolaev) [reply DataConnectors/ThycoticCEF.json:1]: Changed name dataconnector to ThycoticSecretServer

File: DataConnectors/ThycoticCEF.json
Comment (nazang) [reply DataConnectors/ThycoticCEF.json:16]: @AndyNikolaev, Please address the feedback from Jaya. This sample query will not only show results from Thycotic software. Please consider to craft the sample query from base query. For example,  

CommonSecurityLog | where DeviceVendor == \"Thycotic Software\" and DeviceProduct == \"Secret Server\"\n | where Activity contains \"SECRET - VIEW\""

File: DataConnectors/ThycoticCEF.json
Comment (nazang) [reply DataConnectors/ThycoticCEF.json:54]: Please add sharedKey permission. We cannot approve the PR if it is missing.

File: Logos/ThycoticLogo.svg
Comment (nazang) [inline Logos/ThycoticLogo.svg:1]: The logo is not valid, please refer to the previous comment.

File: Workbooks/Images/Logos/ThycoticLogo.svg
Comment (nazang) [inline Workbooks/Images/Logos/ThycoticLogo.svg:1]: The logo file is not valid according to the creation guide. Please check the previous comment.

File: Workbooks/ThycoticWorkbook.json
Comment (nazang) [reply Workbooks/ThycoticWorkbook.json:122]: Same comment as the sample query, please make sure the query only targets logs from Thycotic

File: Workbooks/ThycoticWorkbook.json
Comment (nazang) [inline Workbooks/ThycoticWorkbook.json:192]: Same comment as the sample query, please make sure the query only targets logs from Thycotic

File: Workbooks/ThycoticWorkbook.json
Comment (nazang) [inline Workbooks/ThycoticWorkbook.json:252]: Same comment as the sample query, please make sure the query only targets logs from Thycotic

File: Workbooks/ThycoticWorkbook.json
Comment (nazang) [inline Workbooks/ThycoticWorkbook.json:288]: Same comment as the sample query, please make sure the query only targets logs from Thycotic

**Review by nazang — CHANGES_REQUESTED** [review]
Please update the PR based on Jaya's recommendation.

File: Workbooks/ThycoticWorkbook.json
Comment (AndyNikolaev) [reply Workbooks/ThycoticWorkbook.json]: Add params to query

File: Workbooks/ThycoticWorkbook.json
Comment (AndyNikolaev) [reply Workbooks/ThycoticWorkbook.json]: Add params to query

File: DataConnectors/ThycoticCEF.json
Comment (AndyNikolaev) [reply DataConnectors/ThycoticCEF.json]: Add section

File: Sample Data/CEF/ThycoticSourceData.csv
Comment (AndyNikolaev) [reply Sample Data/CEF/ThycoticSourceData.csv:1]: data in this sample was taken on a test server

File: Dashboards/Images/Logos/ThycoticLogo.svg
Comment (v-jayakal) [inline Dashboards/Images/Logos/ThycoticLogo.svg:8]: Invalid logo.
Ensure raw file of logo does not have any of the following:
style formats
xmlns:xlink
If some properties in the logo have IDs (for e.g. <g id="layer0"...), then set these IDs as GUIDs so that these are uniquely identifiable across all Azure logo assets
Please refer https://github.com/Azure/Azure-Sentinel/blob/e1665d4c4442b430d6847adc65f4b9c25f7b071f/DataConnectors/ReadMe.md

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: Dashboards/Images/Logos/ThycoticLogo.svg
Comment (AndyNikolaev) [reply Dashboards/Images/Logos/ThycoticLogo.svg:4]: Change Logo file

**Comment by AndyNikolaev** [comment]
@v-jayakal @nazang 

I can't understand why when adding a section to a file WorkbooksMetadata.json, the assembly is destroyed 
Section:
{
    "workbookKey": "ThycoticWorkbook",
    "logoFileName": "ThycoticLogo.svg",
    "description": "The Thycotic Secret Server Syslog connector",
    "dataTypesDependencies": [ "CommonSecurityLog" ],
    "dataConnectorsDependencies": [ "ThycoticSecretServer_CEF" ],
    "previewImagesFileNames": ["ThycoticWorkbookWhite.png", "ThycoticWorkbookBlack.png"],
    "version": "1.0",
    "title": "Thycotic Secret Server Workbook",
    "templateRelativePath": "ThycoticWorkbook.json",
    "subtitle": "",
    "provider": "Thycotic"
  },

File: Dashboards/Images/Logos/ThycoticLogo.svg
Comment (v-jayakal) [inline Dashboards/Images/Logos/ThycoticLogo.svg:1]: Provide Guids for **_id_** attributes, remove data-name.
Refer guidelines here https://github.com/Azure/Azure-Sentinel/blob/e1665d4c4442b430d6847adc65f4b9c25f7b071f/DataConnectors/ReadMe.md

Section to be considered:
Ensure raw file of logo does not have any of the following:
cls and style formats
embedded png formats
xmlns:xlink
data-name
If some properties in the logo have IDs (for e.g. <g id="layer0"...), then set these IDs as GUIDs so that these are uniquely identifiable across all Azure logo assets

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Review by v-jayakal — APPROVED** [review]

**Comment by AndyNikolaev** [comment]
> Please update the PR based on Jaya's recommendation.

@nazang

**Review by nazang — APPROVED** [review]

**Review by nazang — CHANGES_REQUESTED** [review]
@AndyNikolaev, could you refresh the branch with the latest master?

There is a error while merging: This branch is out-of-date with the base branch

**Comment by AndyNikolaev** [comment]
@nazang 
pardon me
but always when synchronizing branches, my assembly is destroyed
I can't understand why

File: Logos/ThycoticLogo.svg
Comment (nazang) [inline Logos/ThycoticLogo.svg:1]: @AndyNikolaev, we know the PR is closed but we noticed two additional issues during deployment process:

1. The logo SVG defined a gradient but it’s never used, we suspect that you intended the logo to have some gradient colors. If it is not used, please remove the gradient definition.
<defs>
        <linearGradient x1="57.73" y1="41.36" x2="57.73" y2="48.82" gradientUnits="userSpaceOnUse">
            <stop offset="0.32" stop-color="#82bd41"/>
            <stop offset="1" stop-color="#658d3d"/>
        </linearGradient>
</defs>

2. The preview images included in the PR don't match and don't include the correct images. 
Please see other samples here: https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks/Images/Preview

The preview images should only include the chart.

This issue blocks the deployment of this workbook, please fix it soon.

**Pull request merged**
